src/library_fs.js

```diff
@@ -1204,7 +1204,11 @@ mergeInto(LibraryManager.library, {
       stream.stream_ops.allocate(stream, offset, length);
     },
     mmap: function(stream, buffer, offset, length, position, prot, flags) {
-      // TODO if PROT is PROT_WRITE, make sure we have write access
+      if ((prot & {{{ cDefine('PROT_WRITE') }}}) !== 0
+          && (flags & {{{ cDefine('MAP_PRIVATE')}}}) === 0
```

Comment:
why is private relevant here?

Reply:
According to POSIX spec yes:

"If PROT_WRITE is specified, the application shall ensure that it has opened the file descriptor fildes with write permission unless MAP_PRIVATE is specified in the flags parameter as described below."
http://pubs.opengroup.org/onlinepubs/9699919799/

I'll comment that.

Reply:
Added comment in https://github.com/emscripten-core/emscripten/pull/8633/commits/43fcce7707af84dfe995df60a4fd80e8456e6fad